src/Modules/LearningTasks/Tutor.LearningTasks.Tests/Integration/Learning/TaskProgressTests.cs

```diff
@@ -250,6 +250,78 @@ public void Fails_to_submit_answer_for_missing_task()
         objectResult.StatusCode.ShouldBe(404);
     }
 
+    [Fact]
+    public void OpenSubmission()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+        
+        var result = controller.OpenSubmission(-2, -2, -4);
+
+        result.ShouldBeOfType<OkResult>();
+    }
+
+    [Fact]
+    public void OpenGuidance()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var result = controller.OpenGuidance(-2, -2, -4);
+
+        result.ShouldBeOfType<OkResult>();
+    }
+
+    [Fact]
+    public void OpenExample()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var result = controller.OpenExample(-2, -2, -4);
+
+        result.ShouldBeOfType<OkResult>();
+    }
+
+    [Fact]
+    public void PlayExampleVideo()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var result = controller.PlayExampleVideo(-2, -2, -4, "videoUrl");
+
+        result.ShouldBeOfType<OkResult>();
+    }
+
+    [Fact]
+    public void StopExampleVideo()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var result = controller.PauseExampleVideo(-2, -2, -4, "videoUrl");
+
+        result.ShouldBeOfType<OkResult>();
+    }
+
+    [Fact]
+    public void FinishExampleVideo()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var result = controller.FinishExampleVideo(-2, -2, -4, "videoUrl");
+
+        result.ShouldBeOfType<OkResult>();
+    }
```

Comment:
A Test Suite with 300+ lines of code (or really 200+) should be a signal (code smell) for refactoring.